pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -595,7 +582,13 @@ func (p *PriorityQueue) AddUnschedulableIfNotPresent(logger klog.Logger, pInfo *
 	for plugin := range pInfo.UnschedulablePlugins {
 		metrics.UnschedulableReason(plugin, pInfo.Pod.Spec.SchedulerName).Inc()
 	}
-	if p.moveRequestCycle >= podSchedulingCycle {
+	if p.moveRequestCycle >= pInfo.SchedulingCycle || len(pInfo.UnschedulablePlugins) == 0 {
```

Comment:
Also, this change is still worthwhile for the kube-apiserver failure ([Aldo's comment](https://github.com/kubernetes/kubernetes/pull/119105#issuecomment-1629201087)).
When len(pInfo.UnschedulablePlugins) == 0, it assumes that some unexpected failure happened and put the Pod back to the activeQ/backoffQ. 

I'll rebase + revert unnecesarry changes later!